### PR TITLE
Azure Monitor: Basic Logs modal acknowledgement

### PR DESCRIPTION
**Pull request by jcolladokuri**
**What is this feature?**

This PR adds functionality of a modal so that users always have to acknowledge that they want to switch from Analytics to Basic Logs:

image

this ensures that Editors are aware of the per - query costs before switching to basic

If the modal is not acknowledged, they will remain in Analytics mode.

Fixes #

#84949

### CODE REVIEW

File: public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsManagement.tsx [hunk 1]
```diff
@@ -1,31 +1,52 @@
...
 export function LogsManagement({ query, onQueryChange: onChange }: AzureQueryEditorFieldProps) {
+  const [basicLogsAckOpen, setBasicLogsAckOpen] = useState<boolean>(false);
   return (
     <>
+      <ConfirmModal
+        isOpen={basicLogsAckOpen}
+        title="Basic Logs Queries"
+        body="Are you sure you want to switch to Basic Logs?"
+        description="Basic Logs queries incur cost based on the amount of data scanned. Please acknowledge this panel is subject to per-query costs."
+        confirmText="I Acknowledge"
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsManagement.tsx:18]:
```suggestion
        confirmText="Confirm"
```

Comment (aangelisc) [reply public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsManagement.tsx:18]: I think this might be the default text actually 🤔

**Review by aangelisc — APPROVED** [review]
Excellent work! LGTM 😊

File: public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsManagement.tsx (hunk 1 above)
Comment (jcolladokuri) [reply public/app/plugins/datasource/azuremonitor/components/LogsQueryEditor/LogsManagement.tsx]: I tried removing it to see if there was a default, but the prop is required, so added your suggestion :)

**Pull request merged**